src/jit/importer.cpp

```diff
@@ -13507,19 +13514,31 @@ GenTreePtr Compiler::impAssignStructClassToVar(GenTreePtr op, CORINFO_CLASS_HAND
     unsigned tmpNum = lvaGrabTemp(true DEBUGARG("Return value temp for multireg return."));
     impAssignTempGen(tmpNum, op, hClass, (unsigned) CHECK_SPILL_NONE);
     GenTreePtr ret = gtNewLclvNode(tmpNum, op->gtType);
-#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+
 #ifdef DEBUG
     SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
     eeGetSystemVAmd64PassStructInRegisterDescriptor(hClass, &structDesc);
     // If single eightbyte, the return type would have been normalized and there won't be a temp var.
     // This code will be called only if the struct return has not been normalized (i.e. 2 eightbytes - max allowed.)
     assert(structDesc.passedInRegisters && structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
-#endif // DEBUG
+#endif
+
     // Mark the var to store the eightbytes on stack non promotable.
     // The return value is based on eightbytes, so all the fields need 
     // to be on stack before loading the eightbyte in the corresponding return register.
     lvaTable[tmpNum].lvIsMultiRegArgOrRet = true;
+
+    // TODO-Amd64-Unix-CQ: Right now codegen assumes that tmpNum lcl var is on stack and 
+    // and not lives in a register.  For example, consider Vector3/4 return value of
+    // a call is assigned to tmpNum.  In such a case tmpNum will be of SIMD type and
+    // will be allocated a register unless explicitly marked as DoNotEnregister.
+    //
+    // For now to workaround codegen limitation marking tmpNum as DoNotEnregister.
+    lvaTable[tmpNum].lvDoNotEnregister = true;
```

Comment:
It will not make a material difference now, but I think it is more precise in its application - for example, if we address the unix issue with handling lclVars marked `lvIsMultiRegArgOrRet` (e.g. by reassembling them or something), we may still need to be conservative with lclVars that can be full enregistered, but wouldn't want to pessimize all cases.